File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        // if(s.size())
        unordered_map<char, int> mp;
        for(auto it: s) mp[it]++;
        int p=0;
        int np=0;
        for(auto it: mp){
            int cnt = it.second;
            p+=cnt/2;
            np+=cnt%2;
        }
        return (np<=k && k<=s.size());
    }
};

File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        Map<Character, Integer> mp = new HashMap();
        int np=0;
        for(char c: s.toCharArray()){
            mp.put(c, mp.getOrDefault(c, 0)+1);
        }
        for(int it: mp.values()){
            np+=it%2;
        }
        return np<=k && k<=s.length();
    }
}